Linux/flash

```diff
@@ -207,11 +212,8 @@ while true; do
     esac
   done
 
-  # paranoia sync before umounting
-  echo "Synchronizing Filesystems"
-  sudo sync; sudo sync; sudo sync
-
-  writable=$(grep "${disk}" /etc/mtab | grep -e "\( rw,\|,rw\)")
+  $(sudo test -w "${disk}")
```

Comment:
Yes, I had problems with /etc/mtab; the issue is that an entry will appear in /proc/mounts (which /etc/mtab usually, but not always, symlinks to) if and only if a partition on the SD card gets mounted. However, this will happen only if your distro is set to auto-mount when you insert an SD card (as Ubuntu typically is). My distro isn't setup for that, so the script fails. Thus we need a way to check writability without assuming the disk is mounted.

It's strange that test fails for you. Based on the manpage, it seems like it should succeed if the write bit is set for root (since we run it under sudo). Below is what happens when I try it; could you paste similar output for yourself? We could also use *stat* instead, but it seems like *test -w* should work.

```